test_conformance/subgroups/subhelpers.h

```diff
@@ -352,33 +354,53 @@ struct test
         if (local > LSIZE) local = LSIZE;
 
         // Get the sub group info
-        clGetKernelSubGroupInfoKHR_fn clGetKernelSubGroupInfoKHR_ptr;
-        clGetKernelSubGroupInfoKHR_ptr = (clGetKernelSubGroupInfoKHR_fn)
-            clGetExtensionFunctionAddressForPlatform(
-                platform, "clGetKernelSubGroupInfoKHR");
-        if (clGetKernelSubGroupInfoKHR_ptr == NULL)
+        if (gUseCoreSubgroups)
```

Comment:
Maybe the two paths could be reconciled by passing a structure with function pointers instead of duplicating the code. It's not too bad but it will make maintenance slightly harder. Please verify that the core and KHR enumerants are the same if you want to attempt this though.

You could also have a single utility functions to fill that structure.